### DISCUSSION TITLE
Segregate cache folder based on attributes/flags

### DISCUSSION THREAD

**Discussion by ftrofin** [ccache/ccache#849] [Ideas]
I cross-compile a lot and it would be useful for me if I would be able to have a 'prefix' (e.g. a subfolder) to organize the cache results based on arbitrary criteria (most likely target platform or arch). Currently, all the cache results are mixed in the same cache folder and I can't clean the cache selectively (say delete just the Apple Silicon binaries). So I'd like to specify the 'bucket' in which to store all the cache results.

What are your thoughts on the usefulness of such a feature?

**Comment by jrosdahl** [comment]
Do you envision something similar to setting a variable like `CCACHE_BUCKET=apple_silicon` when compiling Apply Silicon binaries and then running `ccache --clear-bucket apple_silicon`? I guess that would be possible to implement in a backward-compatible way if it's acceptable that `ccache --clear-bucket` will have to scan (read) all results in the cache to find which ones to remove. Doing it more efficiently would require backward-incompatible changes to the cache format which isn't reasonable.

I don't think I have any gut feelings on the usefulness of such a feature, though. Why do you feel the need to clean results of a certain type? Perhaps you could just use different `CCACHE_DIR`s to achieve a similar effect?
